chapters/feature-selection.qmd

```diff
@@ -0,0 +1,525 @@
+
+```{r}
+#| label: fig-selection-sa-resamples
+#| echo: false
+#| out-width: 90%
+#| fig-width: 8
+#| fig-height: 6
+#| fig-cap: The individual SA runs that make up the data in @fig-selection-sa.
+load("../RData/nnet_sa_res.RData")
+
+nnet_sa_res$internal %>%
+  ggplot(aes(Iter, RMSE)) +
+  geom_line() +
+  facet_wrap(~Resample) +
+  labs(x = "SA Iteration", y = "RMSE (Assessment Set)") +
+  geom_hline(yintercept = 0.25, col = "darkgreen", lty = 2)
+```
+
+We should keep in mind that @fig-selection-sa lets us know where, on average, the search should stop, but our actual search result (that uses the entire training set) is more likely to look like what we see in @fig-selection-sa-resamples.
+
+For this particular search, the numerically best result occurred at iteration `r nnet_sa_res$optIter` and the final model contained `r length(nnet_sa_res$optVariables)` predictors where `r xfun::numbers_to_words(sum(grepl("noise", nnet_sa_res$optVariables)))` of the twenty noise columns are still contained in the model. Our resampled estimate of the RMSE, at this iteration, was `r round(mean(nnet_sa_res$internal$RMSE[nnet_sa_res$internal$Iter == nnet_sa_res$optIter]), 3)` (a perfect model would have a value of 0.25). However, the plots of the SA performance profiles indicate a range, probably above 125 iterations, where the results are equivocal.  
+
+## Filter Methods {#sec-filters}
+
+Filters can use any scoring function that quantifies how effectively the predictor predicts the outcome [@kohavi1998wrapper;@duch2006filter]. One method is to use a univariate statistic to evaluate each predictor independently. For example, the area under the ROC curve (@sec-roc) can be computed for a binary classification outcome and a set of numeric predictors. The filter could be applied using these values by pre-defining a threshold for the statistic that defines "important enough." Alternatively, the top $p^*$ predictors can be retained and used in the supervised model. 
+
+The example of using ROC curves raises an interesting question. Should we rank the predictors by their area under the curve or should we convert that analysis to a p-value to see if the area under the curve is greater than 0.5 [@hanley1983method]? Our advice is to do the former. Hypothesis tests and p-values have their uses but the AUC estimates themselves are probably more numerically stable. To factor in uncertainty, we could filter on the lower confidence bound of the AUC.  This is generally true for almost any ranking statistic that has a corresponding p-value. 
+
+Multivariate methods can also be used. These simultaneously compute measures of importance for all predictors. This is usually preferred since univariate statistics can be compromised when the predictors are highly correlated and/or have important interaction effects.  
```

Comment:
Is there a computational argument to be used on whether we use multivariate methods or univariate?

Reply:
It depends on the method. I think my preference is based on efficacy and not efficiency.